Add the refactored NN back into SAM

jatkinson1000 opened the pull request:
Add the [refactored NN for convection parameterisation](https://github.com/m2lines/convection-parameterization-in-CAM) back into SAM.

- [ ] Update the README so it is clear what this repo is and how it differs from the original 
- [ ] Adapt SAM to use our implementation
- [ ] Switch between Janni and new with namelist flag
- [x] Import our source code from Convection repo
- [x] Check it compiles and runs OK
- [ ] Compare outputs (and document in README how this can be manually diff'd)

jatkinson1000 committed:
modify SAM to use our interface routines, with a logical switch provided in namelist. WIP - needs our NN source adding.

dorchard commented:
Great. First two tasks look done? Can we do a submodule for task 3 (to import from the convection repo)?

jatkinson1000 commented:
> Great. First two tasks look done? Can we do a submodule for task 3 (to import from the convection repo)?

Mostly, just need to check and then I'll tick off.
Considered a submodule, but just want the source files rather than the whole repo, and preferably adjacent to the code rather than in a subdirectory for building purposes... Can discuss later today perhaps?

raehik committed:
submodule refactored neural net

jatkinson1000 committed:
Add files from Cheyenne to build.

jatkinson1000 committed:
Add CMake files for building.

raehik committed:
begin Nix build

raehik committed:
Nix flake: fix install phase

raehik committed:
main: temporarily comment out refs to new NN

raehik committed:
Nix flake: tweaks

raehik committed:
setparm: abs->rel path

raehik committed:
cmake: include refactored neural net dir

Needs some more edits to properly build: a CMakeLists for refactored NN,
and changes to subroutine names or different importing so that names
don't clash.

raehik commented:
Some conflicting function/subroutine names between the original and refactored NN code made the "switching" awkward. @dorchard suggested we gate with CPP instead -- looking into it next.
